### DISCUSSION TITLE
(Linux: ubuntu) How to open another application from tauri

### DISCUSSION THREAD

**Discussion by devhimanshuhc** [tauri-apps/tauri#8850] [Q&A]
So, i have an application built with reactJS, and i want to have a button, and when click on that button i want to trigger vs code to open.  tried the following code for but it threw me an error on clicking the button.
```js
  const openVsCode = async () => {
    try {
      await open("/snap/bin/code");
    } catch (error) {
      console.error("Error opening VS Code:", error); // Handle any errors
      // Optionally provide user feedback or alternative actions
    }
  };
```
but it threw error to me:

```
"shell error: failed to open: Scoped command argument at position 0 was found, but failed regex validation ^((mailto:\\w+)|(tel:\\w+)|(https?://\\w+)).+"
```
I ckecked the path of vs code using `whereis code` and `which code` and its `/snap/bin/code`. can anyone help me out with it?

![Screenshot from 2024-02-13 18-51-36](https://github.com/tauri-apps/tauri/assets/80395823/84c456d9-5997-42b6-bc1b-05f342fb8056)

**Comment by soopermodel** [comment]
same issue here

**Comment by mehmedaltug** [comment]
you can use std::process::Command

for example:
Command::new("/snap/bin/code").arg("/home");

more info: [https://doc.rust-lang.org/std/process/struct.Command.html](https://doc.rust-lang.org/std/process/struct.Command.html)

**Comment by comTg** [comment]
same issue here, 
Still not fixed?
